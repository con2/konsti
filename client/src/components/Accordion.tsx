import React, { FC, ReactElement } from 'react';
import { useTranslation } from 'react-i18next';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import styled from 'styled-components';

export interface Props {
  text: string;
  title: string;
  buttonText: string;
}

export const Accordion: FC<Props> = (props: Props): ReactElement => {
  const { text, title, buttonText } = props;

  const [open, setOpen] = React.useState<boolean>(false);

  const { t } = useTranslation();

  const onClick = (): void => {
    setOpen(!open);
  };

  const splitTextRows = (text: string): ReactElement[] => {
    const rows = t(text).split('\n');
    return rows.map((row) => <p key={row}>{row}</p>);
  };

  return (
    <div className='accordion'>
      {open && (
        <>
          <AccordionToggle onClick={() => onClick()}>
            <AccordionIcon icon='angle-up' />
            {t(`${buttonText}`)}
          </AccordionToggle>
          <AccordionContent>
            <h3>{t(`${title}`)}</h3>
            <div>{splitTextRows(text)}</div>
          </AccordionContent>
        </>
      )}
      {!open && (
        <>
          <AccordionToggle onClick={() => onClick()}>
            <AccordionIcon icon='angle-down' />
            <span>{t(`${buttonText}`)}</span>
          </AccordionToggle>
        </>
      )}
    </div>
  );
};

const AccordionContent = styled.div`
  box-shadow: 0 1px 0 rgba(9, 30, 66, 0.25);
  border: 1px solid ${(props) => props.theme.borderInactive};
  border-radius: 3px;
  padding: 0 10px;
`;

const AccordionToggle = styled.button`
  padding: 6px 10px;
`;

const AccordionIcon = styled(FontAwesomeIcon)`
  margin: 0 10px 0 0;
  font-size: 18px;
`;
