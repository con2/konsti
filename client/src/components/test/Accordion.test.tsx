import React from 'react';
import { shallow } from 'enzyme';
import { Accordion, Props } from '../Accordion';

const title = 'test title';
const text = 'test text';
const buttonText = 'test button text';

describe('Loading', () => {
  it('should render correctly', () => {
    const props: Props = { title, text, buttonText };
    const component = shallow(<Accordion {...props} />);
    expect(component).toMatchSnapshot();
  });
});
