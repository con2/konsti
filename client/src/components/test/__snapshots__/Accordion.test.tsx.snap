// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render correctly 1`] = `
<div
  className="accordion"
>
  <Accordion__AccordionToggle
    onClick={[Function]}
  >
    <Accordion__AccordionIcon
      icon="angle-down"
    />
    <span>
      test button text
    </span>
  </Accordion__AccordionToggle>
</div>
`;
